hazelcast/src/main/java/com/hazelcast/nio/tcp/AbstractSelectionHandler.java

```diff
@@ -46,15 +47,43 @@ public AbstractSelectionHandler(TcpIpConnection connection, IOSelector ioSelecto
         this.initialOps = initialOps;
     }
 
+    public void migrate(final IOSelector newIOSelector) {
+        if (ioSelector == newIOSelector) {
+            return;
+        }
+
+        ioSelector.addTask(new Runnable() {
+            @Override
+            public void run() {
+                unregisterOp(initialOps);
+                ioSelector = newIOSelector;
```

Comment:
I'm missing a selectionKey.cancel:

http://stackoverflow.com/questions/13867133/is-there-a-way-to-de-register-a-selector-on-a-socket-channel